ee/debug/checkups/performance.go

```diff
@@ -21,9 +28,20 @@ func (p *perfCheckup) Run(ctx context.Context, _ io.Writer) error {
 	p.data = make(map[string]any)
 	stats, err := performance.CurrentProcessStats(ctx)
 	if err != nil {
+		p.status = Erroring
 		return fmt.Errorf("gathering performance stats: %w", err)
 	}
 
+	memOver := stats.MemInfo.GoMemUsage > golangMemUsageThreshold || stats.MemInfo.NonGoMemUsage > nonGolangMemUsageThreshold
+	cpuOver := stats.CPUPercent > cpuPercentThreshold
+	if cpuOver && memOver {
+		p.status = Failing
+	} else if cpuOver || memOver {
+		p.status = Warning
+	} else {
+		p.status = Passing
+	}
```

Comment:
Logic feels weird.

I would say that if we exceed _either_ threshold we're Failing. 

If we want a warning state, we could define those thresholds.